orderbook/orderbook.go

```diff
@@ -1,122 +1,149 @@
 package orderbook
 
 import (
+	"reflect"
 	"sync"
 
 	"github.com/republicprotocol/republic-go/dispatch"
 	"github.com/republicprotocol/republic-go/order"
 )
 
 type Syncer interface {
-	Open(message *Message) error
-	Match(message *Message) error
-	Confirm(message *Message) error
-	Release(message *Message) error
-	Settle(message *Message) error
+	Open(entry Entry) error
+	Match(entry Entry) error
+	Confirm(entry Entry) error
+	Release(entry Entry) error
+	Settle(entry Entry) error
 	Cancel(id order.ID) error
 }
 
 // Broadcaster is the subject in the observer design pattern
 type Broadcaster interface {
-	Subscribe(id string, queue dispatch.MessageQueue) error
-	Unsubscribe(id string)
+	Subscribe(ch interface{}) error
+	Unsubscribe(ch interface{})
+	Close()
 }
 
 // An Orderbook is responsible for store the historical orders both in cache
 // and in disk. It also streams the newly received orders to its subscriber.
 type Orderbook struct {
-	orderBookCache OrderBookCache
-	orderBookDB    OrderBookDB
-	splitter       dispatch.Splitter
+	cache    Cache
+	database Database
+	splitter dispatch.Splitter
+	splitCh  chan Entry
 }
 
-// NewOrderBook creates a new OrderBook with the given logger and splitter
-func NewOrderBook(maxConnections int) *OrderBook {
-	return &OrderBook{
-		orderBookCache: NewOrderBookCache(),
-		orderBookDB:    OrderBookDB{},
-		splitter:       dispatch.NewSplitter(maxConnections),
+// NewOrderbook creates a new Orderbook with the given logger and splitter
+func NewOrderbook(maxConnections int) Orderbook {
+	splitter := dispatch.NewSplitter(maxConnections)
+	splitCh := make(chan Entry)
+	go splitter.Split(splitCh)
+
+	return Orderbook{
+		cache:    NewCache(),
+		database: Database{},
+		splitter: splitter,
+		splitCh:  splitCh,
 	}
 }
 
 // Subscribe will start listening to the orderbook for updates.
-func (orderBook OrderBook) Subscribe(id string, queue dispatch.MessageQueue) error {
+func (orderbook Orderbook) Subscribe(ch interface{}) error {
```

Comment:
The channel should be typed here. There's not much point leaving it as `interface{}` instead of `chan Entry`. The documentation for this method should also explicit state that the channel must not be closed until after `Unsubscribe` is called.